tests/test_benchmark_constraints.py

```diff
@@ -104,6 +112,92 @@ def test_benchmark_minimal_renewable_share_constraint(self, margs):
             renewable_shares[use_case[0]] < renewable_shares[use_case[1]]
         ), f"The renewable share of the scenario with the constraint is not higher than the one without, so the test does not make sense."
 
+    @pytest.mark.skipif(
+        EXECUTE_TESTS_ON not in (TESTS_ON_MASTER),
+        reason="Benchmark test deactivated, set env variable "
+        "EXECUTE_TESTS_ON to 'master' to run this test",
+    )
+    @mock.patch("argparse.ArgumentParser.parse_args", return_value=argparse.Namespace())
+    def test_benchmark_maximum_emissions_constraint(self, margs):
+        r"""
+        Tests the maximum emissions constraint in a system with PV, DSO and a diesel generator.
+        The şystem defined in `\Constraint_maximum_emissions_None` does not have maximum emissions constraint,
+        while the system defined in `\Constraint_maximum_emissions_low` has a low maximum emissions constraint of 800 kgCO2eq/a.
+        A third system, `\Constraint_maximum_emissions_low_grid_RE_100`, includes a renewable share in the grid of 100 %.
+
+        The following checks are made:
+        - total emissions of energy system <= maximum emissions constraint
+            (for Constraint_maximum_emissions_low and Constraint_maximum_emissions_low_grid_RE_100)
+        - total emissions of case without constraint > total emissions of case with constraint
+        - specific emissions eleq of case without constraint > specific emissions eleq of case with constraint
+        - optimized added pv capacity lower for case without constraint than for case with constraint
+        - optimized added capacity of diesel higher for case with 100 % RE share in grid
```

Comment:
Why would this be higher? Because diesel is very cheap and would be preffered over PV? I would have thought that only grid consumption increases in this scenario.

Reply:
Now I'm sure of it anymore, either. 
I've checked the results of `Constraint_maximum_emissions_low_grid_RE_100`. The lcoe of the diesel generator is with 0.12 lower than the `energy_price` of the grid (0.3). 
Therefore, diesel is preferred (up to maximum emissions limit). 

When the grid has emissions, too, they are far lower than the ones from diesel. Therefore, by using the grid more (means installing less or no diesel power) you can prevent the high costs resulting from higher PV capacity. 

But as this is thought around a lot of corners, I'll might delete this check. 
Definitely adding the check that grid consumption is used more (cause this is also an effect of my above explanations).